x-pack/plugins/infra/public/components/waffle/node_context_menu.tsx

```diff
@@ -90,7 +90,9 @@ export const NodeContextMenu = injectUICapabilities(
             }
           : undefined;
 
-      const uptimeUrl = node.ip
+      const showUptimeLink =
```

Comment:
This is possibly subjective but could we re-arrange this a bit? I like having the clear boolean intermediates like  you've added with `showUptimeLink`, so maybe we could make this section look like this:
```ts
const nodeLogsUrl = getNodeLogsUrl({
  nodeType,
  nodeId: node.id,
  time: timeRange.to,
});
const nodeDetailUrl = getNodeDetailUrl(
  // etc
);
const apmTracesUrl = {
  // etc
};
const uptimeUrl = {
  // etc
};

const showNodeLogsLink = node.id && uiCapabilities.logs.show;
const showNodeDetailLink = Boolean(node.id);
const showApmTracesLink = uiCapabilities.apm && uiCapabilities.apm.show;
const showUptimeLink =
  [InfraNodeType.pod, InfraNodeType.container].includes(nodeType) || node.ip;
```

Then the `items` creation below can still be the `...(showNodeLogsLink ? [nodeLogsUrl] : [])` it is now or even just a `showNodeLogsLink && nodeLogsUrl` for each link, with a `.filter(link => link)` at the end, whatever you think is clearer/prefer. 

Is that outside the scope of this fix? The logic in here was just getting a bit hard to follow.

Reply:
I just pushed an update... better or worse?

Reply:
Much better, thanks!